### PR TITLE
Added the argument to set container to use python 3.7 in api Dockerfile

### PR DESCRIPTION
**Pull request by jainesh99**
Hello all,
This is my first pull request for this project, I have set the container for the back end (api) to use python 3.7.

Please have a look and let me know if there is anything else required. 

Thank You
Jainesh J Hira

### CODE REVIEW

**Comment by nischalhp** [comment]
Hello Jainesh,

Looks like the update in the container is failing to build the application ( as expected ) . Could you also take a look at why the application build is failing and submit all the changes accordingly? 

Also, congratualtions on your first PR to this project :) 

㊗️

**Comment by jainesh99** [comment]
Sure will do so, thank you for the feedback.

**Comment by amritpurshotam** [comment]
Hey @jainesh99, sorry about this. I think Jade is working on it. Can you hold off until @jaderabbit confirms where she is on this?

**Comment by jainesh99** [comment]
Hi @amritpurshotam , no problem at all, I'll look at another task.

**Comment by jainesh99** [comment]
Should I close this pull request to avoid confusion?

**Comment by avishkar58** [comment]
Let's wait to hear from Jade, I'm not entirely sure what the scope of her
changes are.

**Comment by jaderabbit** [comment]
Hi All!

So, to upgrade to python 3, there is a bit of work to be done!
We need to update the imports to be python compatible and a few other nitbits

I have a partially implemented branch (So I started manually doing the upgrade) and then got tired. 
I can push my branch and @jainesh99  can take over?

**Comment by jainesh99** [comment]
Hi @jaderabbit ,

Sure you can push your branch and I can have a look at it and try and see what I can do :)

**Comment by jaderabbit** [comment]
The branch: https://github.com/deep-learning-indaba/Baobab/tree/feature/upgrade-circleci-py3

You'll see if you run tests, they're MOSTLY failing

**Comment by jainesh99** [comment]
Hi All,

Please can you review this code, the only changes which I am not a bit comfortable with is the copying of the routes.py into the __init__.py, I have also checked in a new requirements.txt which is what I used to test with python3.

**Comment by jainesh99** [comment]
Resolves #454

File: api/requirements_new.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
+alembic==1.3.3
```
Comment (amritpurshotam) [inline api/requirements_new.txt]: Can we not replace the current requirements file?

**Comment by avishkar58** [comment]
Lets hold off on merging this, since there is a high risk of subtle bugs and we're currently testing the EEML functionality. Once their application form is live in production (ETA next week) we can merge this in and test it thoroughly.

**Comment by amritpurshotam** [comment]
Sure that makes sense. I was also thinking we should wait til after the EEML deployment.

File: api/requirements_new.txt (hunk 1 above)
Comment (jainesh99) [reply api/requirements_new.txt]: Yes sure we can.

**Comment by jainesh99** [comment]
Sure no problem, will hold off this for now, I think some extensive testing will be needed.

**Comment by jainesh99** [comment]
Is it fine if I add the do not merge label to the ticket for now.

**Comment by avishkar58** [comment]
> Is it fine if I add the do not merge label to the ticket for now.

Yes please! thanks @jainesh99  :)

**Comment by jainesh99** [comment]
I have updated the latest code to be compatible with python 3.7. The only caveat is that all the code from routes.py had to be pasted into the __init__.py which is not ideal!

@avishkar58 , @jaderabbit  please can you have a look, also if you have any ideas how to get around this problem.

**Pull request closed without merging**